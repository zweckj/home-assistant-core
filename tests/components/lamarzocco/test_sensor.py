"""Tests for La Marzocco sensors."""
from unittest.mock import MagicMock

import pytest
from syrupy import SnapshotAssertion

from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

pytestmark = pytest.mark.usefixtures("init_integration")


SENSORS = (
    "drink_statistics_coffee",
    "drink_statistics_coffee",
    "shot_timer",
    "current_coffee_temperature",
    "current_steam_temperature",
)


async def test_sensors(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the La Marzocco sensors."""

    serial_number = mock_lamarzocco.serial_number

    for sensor in SENSORS:
        state = hass.states.get(f"sensor.{serial_number}_{sensor}")
        assert state
        assert state == snapshot(name=f"{serial_number}_{sensor}-sensor")

        entry = entity_registry.async_get(state.entity_id)
        assert entry
        assert entry.device_id
        assert entry == snapshot(name=f"{serial_number}_{sensor}-entry")
