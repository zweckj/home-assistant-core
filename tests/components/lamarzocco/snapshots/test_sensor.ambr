# serializer version: 1
# name: test_sensors[GS01234_current_coffee_temperature-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.gs01234_current_coffee_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': 'mdi:thermometer',
    'original_name': 'Current coffee temperature',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'current_temp_coffee',
    'unique_id': 'GS01234_current_temp_coffee',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '째C'>,
  })
# ---
# name: test_sensors[GS01234_current_coffee_temperature-sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'GS01234 Current coffee temperature',
      'icon': 'mdi:thermometer',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '째C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.gs01234_current_coffee_temperature',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '93',
  })
# ---
# name: test_sensors[GS01234_current_steam_temperature-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.gs01234_current_steam_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': 'mdi:thermometer',
    'original_name': 'Current steam temperature',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'current_temp_steam',
    'unique_id': 'GS01234_current_temp_steam',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '째C'>,
  })
# ---
# name: test_sensors[GS01234_current_steam_temperature-sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'GS01234 Current steam temperature',
      'icon': 'mdi:thermometer',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '째C'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.gs01234_current_steam_temperature',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '113',
  })
# ---
# name: test_sensors[GS01234_drink_statistics_coffee-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.gs01234_drink_statistics_coffee',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:chart-line',
    'original_name': 'Drink statistics coffee',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'drink_stats_coffee',
    'unique_id': 'GS01234_drink_stats_coffee',
    'unit_of_measurement': 'drinks',
  })
# ---
# name: test_sensors[GS01234_drink_statistics_coffee-sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'GS01234 Drink statistics coffee',
      'icon': 'mdi:chart-line',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': 'drinks',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.gs01234_drink_statistics_coffee',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '13',
  })
# ---
# name: test_sensors[GS01234_shot_timer-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': <EntityCategory.DIAGNOSTIC: 'diagnostic'>,
    'entity_id': 'sensor.gs01234_shot_timer',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <SensorDeviceClass.DURATION: 'duration'>,
    'original_icon': 'mdi:timer',
    'original_name': 'Shot timer',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'shot_timer',
    'unique_id': 'GS01234_shot_timer',
    'unit_of_measurement': <UnitOfTime.SECONDS: 's'>,
  })
# ---
# name: test_sensors[GS01234_shot_timer-sensor]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'duration',
      'friendly_name': 'GS01234 Shot timer',
      'icon': 'mdi:timer',
      'state_class': <SensorStateClass.MEASUREMENT: 'measurement'>,
      'unit_of_measurement': <UnitOfTime.SECONDS: 's'>,
    }),
    'context': <ANY>,
    'entity_id': 'sensor.gs01234_shot_timer',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '0',
  })
# ---
